src/NimBLERemoteCharacteristic.cpp

```diff
@@ -132,51 +126,36 @@ bool NimBLERemoteCharacteristic::retrieveDescriptors(const NimBLEUUID* uuidFilte
  */
 NimBLERemoteDescriptor* NimBLERemoteCharacteristic::getDescriptor(const NimBLEUUID& uuid) const {
     NIMBLE_LOGD(LOG_TAG, ">> getDescriptor: uuid: %s", uuid.toString().c_str());
-    NimBLERemoteDescriptor* pDsc      = nullptr;
-    size_t                  prev_size = m_vDescriptors.size();
+    NimBLETaskData taskData(const_cast<NimBLERemoteCharacteristic*>(this));
+    desc_filter_t  filter = {nullptr, &uuid, &taskData};
+    NimBLEUUID     uuidTmp;
 
     for (const auto& it : m_vDescriptors) {
         if (it->getUUID() == uuid) {
-            pDsc = it;
+            filter.dsc = it;
```

Comment:
Since we are in here, could you change the variable name from `it` to `dsc` since that is not an iterator.